app/services/fact_importer.rb

```diff
@@ -111,7 +115,11 @@ def normalize(facts)
     facts.keep_if { |k, v| v.present? }
   end
 
+  def db_facts_as_relation
+    host.fact_values.eager_load(:fact_name).where("fact_names.type = '#{fact_name_class}'")
+  end
+
   def db_facts
-    @db_facts ||= host.fact_values.eager_load(:fact_name).where("fact_names.type = '#{fact_name_class}'").index_by(&:name)
+    @db_facts ||= db_facts_as_relation.index_by(&:name)
```

Comment:
But this method is only called once, I don't think this is worth changing to pluck, we need to create hash from it anyway (`index_by` creates a hash).